src/hooks/useDialog/useDialog.tsx

```diff
@@ -0,0 +1,121 @@
+import MessageDialog, { MessageDialogProps } from '@/components/Dialogs/MessageDialog'
+import { createId } from '@/utils/createId'
+import React, { useCallback, useEffect, useState, useContext, useMemo } from 'react'
+import { TransitionGroup } from 'react-transition-group'
+
+type DialogRendererProps = {
+  component: React.FunctionComponent
+}
+
+interface DialogsContainerProps {
+  dialogs: Record<string, React.FunctionComponent>
+  component?: React.ComponentType
+  container?: Element
+}
+
+const DialogsRenderer = ({ component, ...rest }: DialogRendererProps) => component(rest)
+
+export const DialogsContainer = ({ dialogs }: DialogsContainerProps) => {
+  return (
+    <TransitionGroup>
+      {Object.keys(dialogs).map((key) => (
+        <DialogsRenderer key={key} component={dialogs[key]} />
+      ))}
+    </TransitionGroup>
+  )
+}
+
+type DialogContextValue = {
+  openDialog: (key: string, dialog: React.FunctionComponent<{ in?: boolean }>) => void
+  closeDialog: (key: string) => void
+}
+const DialogContext = React.createContext<undefined | DialogContextValue>(undefined)
+DialogContext.displayName = 'DialogsContext'
+
+type DialogsProviderProps = {
+  container?: Element
+}
+
+export const DialogProvider: React.FC<DialogsProviderProps> = ({ container, children }) => {
+  const [dialogs, setDialogs] = useState<Record<string, React.FC>>({})
+
+  const openDialog = useCallback(
+    (key: string, dialog: React.FC) =>
+      setDialogs((dialogs) => ({
+        ...dialogs,
+        [key]: dialog,
+      })),
+    []
+  )
+
+  const closeDialog = useCallback(
+    (key: string) =>
+      setDialogs((dialogs) => {
+        if (!dialogs[key]) {
+          return dialogs
+        }
+        const newDialogs = { ...dialogs }
+        delete newDialogs[key]
+        return newDialogs
```

Comment:
Should be simpler

```suggestion
        const { [key]: _, ...filteredDialogs } = dialogs
        return filteredDialogs
```

Reply:
nice! Forgot that I can use computed property like this